lib/wasi/src/runtime/mod.rs

```diff
@@ -34,6 +38,13 @@ where
 
     fn package_resolver(&self) -> Arc<dyn PackageResolver + Send + Sync>;
 
+    /// A cache for compiled modules.
+    ///
+    /// Caching is disabled by default.
+    fn module_cache(&self) -> Arc<dyn ModuleCache + Send + Sync> {
+        Arc::new(module_cache::Disabled)
```

Comment:
There's no guarantee that the returned module cache will be saved by the caller and reused everywhere, so returning a new instance of the default cache every time means the cached modules won't actually be shared or persisted.

The best way to avoid footguns is to not provide a default implementation and force the implementor to handle things themselves.